.github/scripts/check_copyright_years.py

```diff
@@ -0,0 +1,129 @@
+            break
+        yield commit
+
+def get_relevant_files_from_diff_index(diff_index):
+    """
+    Get all filepaths that are added, modified, or renamed in a DiffIndex.
+    """
+    for diff in iter_change_types(diff_index, ("A", "M", "R")):
+        yield diff.b_path
+
+def get_copyright_years(repo, commit, path):
+    content = repo.git.show(f"{commit.hexsha}:{path}")
+    lines = content.split("\n")
+
+    for lineno, line in enumerate(lines, start=1):
+        if SPDX_MAGIC in line:
+            matches = YEAR_RE.findall(line)
+            if not matches:
+                # XXX: We don't error on "poorly formatted" SPDX headers, as our
+                #      current check also triggers on things like our SPDX_MAGIC
+                #      definition.
+                continue
+            yield int(matches[-1]), lineno
+
+def is_valid_commit_file(repo, commit, path) -> bool:
+    """
+    Checks whether a specific file in a specific commit has up-to-date copyright
+    notices.
+    """
+    commit_year = commit.authored_datetime.year
+    max_year, line_no = max(get_copyright_years(repo, commit, path))
+
+    if max_year != commit_year:
+        print(f"{commit.hexsha} {path}:{line_no} Unexpected copyright year. Expected {commit_year}, got {max_year}.", file=sys.stderr)
+
+    return True
+
+
+def is_valid_commit(repo, commit, seen_files) -> Tuple[bool, Set[str]]:
+    """
+    Check whether all files touched in given commit have up-to-date copyright
+    notices. Ignores `seen_files` to make sure only the oldest commits introducing
+    and issue get listed.
+
+    Returns a tuple of (everything_ok, checked_files)
+    """
+    if not commit.parents:
+        print("Ignoring root commit..", file=sys.stderr)
+        return True, set()
+
+    parent = commit.parents[0]
+    files = set(get_relevant_files_from_diff_index(parent.diff(commit)))
+    files -= seen_files
+    files = {f for f in files if not f.startswith(IGNORE_STARTSWITH)}
+
+    all_valid = True
+    for file in files:
+        all_valid &= is_valid_commit_file(repo, commit, file)
+
+    return all_valid, files
+
+
+def main(merge_base_branch):
+    repo = git.Repo(".")
+
+    # Get the last common ancestor of the current branch and the merge base
+    current_branch = repo.active_branch
+    common_ancestor_hexdigest = repo.git.merge_base(merge_base_branch, current_branch.name)
+    common_ancestor = repo.commit(common_ancestor_hexdigest)
+
+    if current_branch.commit.binsha == common_ancestor.binsha:
+        exit(f"No commits on top of base branch '{merge_base_branch}'. Empty PR?")
+
+    # Get the commits that are in the current branch but not in the merge target
+    pr_commits = get_pr_commits(repo, common_ancestor)
+
+    # Check if the year in the license header is correct for each file in each commit
+    all_valid = True
+    seen = set()
+    for commit in pr_commits:
+        (commit_valid, files_in_commit) = is_valid_commit(repo, commit, seen)
```

Comment:
You can also just ignore this one, as both with and without brackets is valid. You unpack _without_ brackets in line 65, so removing the brackets here is more uniform.
```suggestion
        commit_valid, files_in_commit = is_valid_commit(repo, commit, seen)
```